contracts/libraries/WeightedOracleLibrary.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0 <0.8.0;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+
+/// @title Weighted Oracle library
+/// @notice Provides functions to integrate with different tier oracles of the same V3 pair
+library WeightedOracleLibrary {
+
+    struct TimeWeightedObservation {
+        int24 timeWeightedAverageTick;
+        uint256 timeWeightedHarmonicMeanLiquidity;
```

Comment:
using "mean" here and "average" for tick, maybe consistently use one or other. average is more human readable

Reply:
also i think this should be a uint128